lib/collection/property-list.js

```diff
@@ -215,7 +220,24 @@ _.assign(PropertyList.prototype, /** @lends PropertyList.prototype */ {
             if (predicate.apply(context, arguments)) {
                 if ((index = item[this._postman_listIndexKey]) && (index = String(index))) {
                     this._postman_listIndexCaseInsensitive && (index = index.toLowerCase());
-                    delete this.reference[index];
+
+                    if (this._postman_listAllowsMultipleValues && _.isArray(this.reference[index])) {
+                        // since we have an array of multiple values, remove only the value for which the
+                        // predicate returned truthy. If the array becomes empty, just delete it.
+                        _.remove(this.reference[index], function (each) {
+                            return each === item;
+                        });
+
+                        // If the array becomes empty, remove it
+                        (this.reference[index].length === 0) && (delete this.reference[index]);
```

Comment:
No need for unnecessary lodash usage, it does additional things. We're guaranteed to have an array because of the condition above, so directly checking the length is safe and extremely fast.